system/security/shared/caching.cpp

```diff
@@ -435,17 +432,16 @@ void CPermissionsCache::add(ISecUser& sec_user)
 #ifdef _DEBUG
     DBGLOG("CACHE: CPermissionsCache Adding cached user %s", username);
 #endif
-    m_userCache[username] = new CachedUser(sec_user.clone());
+    m_userCache[username] = new CachedUser(LINK(&sec_user));
```

Comment:
The only danger of this change is if there are any local class instances passed in.  I don't think there are, but the current way the files are structured means I think it is hard to protect yourself against it.